### PR TITLE
add local testing of json schema generation

### PR DESCRIPTION
**Pull request by surajssd**

### CODE REVIEW

**Comment by ashetty1** [comment]
Tried it. WFM

**Comment by ashetty1** [comment]
Tested it and it works

**Comment by surajssd** [comment]
@ashetty1 thanks!

**Comment by surajssd** [comment]
@kadel ping

File: Makefile [hunk 1]
```diff
@@ -86,7 +86,7 @@ endif
 
 # Run all tests
 .PHONY: test
-test: test-dep validate test-unit
+test: test-dep validate test-unit test-jsonschema-generation
```
Comment (cdrage) [inline Makefile]: IMO this should not be included.

This suddenly uses a Docker container. 

Similar to how we have integration tests / unit tests / cmd tests separated into their separate commands.

Comment (surajssd) [reply Makefile]: Is it not safe to assume that the kedge developers have docker installed locally?

Comment (cdrage) [reply Makefile]: no, it's not safe, and tests should be ran separately... testing json-schema-generation should be separate just like test-unit, test-ci, test-cmd, etc.

Comment (surajssd) [reply Makefile]: done changes as requested!

**Comment by surajnarwade** [comment]
if I run it locally, I get following error:

```
$ make test-jsonschema-generation 
docker run -v `pwd`/pkg/spec/spec.go:/data/spec.go:ro,Z surajd/kedgeschema
+ curl https://raw.githubusercontent.com/kedgeproject/json-schema-generator/master/scripts/k8s-release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    12  100    12    0     0     12      0  0:00:01 --:--:--  0:00:01    22
+ curl -O https://raw.githubusercontent.com/kubernetes/kubernetes/release-1.7/api/openapi-spec/swagger.json
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1776k  100 1776k    0     0    423      0  1:11:40  1:11:36  0:00:04 1023k
+ curl -O https://raw.githubusercontent.com/kedgeproject/kedge/master/pkg/spec/spec.go
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Warning: Failed to create the file spec.go: Read-only file system
 25  7696   25  1925    0     0   1925      0  0:00:03 --:--:--  0:00:03  3493
curl: (23) Failed writing body (0 != 1925)
Makefile:125: recipe for target 'test-jsonschema-generation' failed
make: *** [test-jsonschema-generation] Error 23
```

**Comment by surajnarwade** [comment]
using `sudo` in travis will work (for docker)

**Comment by surajssd** [comment]
@surajnarwade 

>if I run it locally, I get following error:

for some reason curl seemed to fail in there

>using sudo in travis will work (for docker)

yes added sudo in front of docker command and also added sudo: required in travis config file

File: Makefile [hunk 2]
```diff
@@ -118,3 +118,9 @@ vendor-update:
 	glide update -v
 	# Vendors OpenShift and its dependencies
 	./scripts/vendor-openshift.sh
+
+# Test if the changed spec.go is valid and JSONSchema can be generated out of it
+.PHONY: test-jsonschema-generation
+test-jsonschema-generation:
+	sudo docker run -v `pwd`/pkg/spec/spec.go:/data/spec.go:ro,Z surajd/kedgeschema
```
Comment (cdrage) [inline Makefile]: having anything "sudo" within Makefile is really bad...

Please remove sudo! If a user wishes to run test-jsonschema-generation they can use `make test-jsonschema-generation`

File: .travis.yml [hunk 3]
```diff
@@ -1,5 +1,7 @@
 dist: trusty
 
+sudo: required
```
Comment (cdrage) [inline .travis.yml]: will need to add docker as a service to here too:
```
sudo: required

services:
  - docker
```

File: Makefile (hunk 2 above)
Comment (surajssd) [reply Makefile]: resolved!

File: .travis.yml (hunk 3 above)
Comment (surajssd) [reply .travis.yml]: done, thanks!

**Comment by cdrage** [comment]
Other than SemaphoreCI test failing, LGTM!

**Pull request merged**